memo.rb

```diff
@@ -0,0 +1,108 @@
+# frozen_string_literal: true
+
+require 'sinatra'
+require 'sinatra/reloader'
+require 'json'
+
+require_relative 'memo_class'
+
+helpers do
+  def h(text)
+    Rack::Utils.escape_html(text)
+  end
+
+  def params_validation(path, title, content)
+    redirect to(path) if title.empty? || content.empty?
+  end
+
+  def all_memos
+    File.open('memos.json') do |file|
+      FileTest.empty?('memos.json') ? nil : JSON.parse(file.read, symbolize_names: true)
+    end
+  end
+
+  def find_memo(memos, id)
+    memos.find { |memo| memo[:id] == id }
+  end
+
+  def save_memos(memos)
+    File.open('memos.json', 'w') { |file| JSON.dump(memos, file) }
+  end
+end
+
+get '/memos' do
+  @title = 'メモ一覧'
+  @memos = all_memos
+
+  erb :index
+end
+
+get '/memos/new' do
+  @title = 'メモ作成'
+
+  erb :new
+end
+
+post '/memos' do
+  params_validation('/memos/new', params[:title], params[:content])
+
+  memo = Memo.new(params[:title], params[:content])
+  id = memo.id
+  title = memo.title
+  content = memo.content
+  created_at = memo.created_at
+  updated_at = memo.updated_at
+
+  File.open('memos.json') do |file|
+    # メモ格納ファイルが空の場合は、空配列を用意する
+    memos = FileTest.empty?('memos.json') ? [] : JSON.parse(file.read, symbolize_names: true)
+    memo_hash = { id: id, title: title, content: content, created_at: created_at, updated_at: updated_at }
```

Comment:
`Memo` くらすに `#to_hash` というメソッドを追加すれば

```suggestion
    memo_hash = memo.to_hash
```

という具合にかけて、コントローラーがシンプルになると思います。

Reply:
メモクラスに`to_hash`メソッドを追加してメモのハッシュを返す様にしました。
https://github.com/Kassy0220/memo_app/blob/60082615e7dfc0bea9ac794a920359c13ff61074/memo.rb#L18-L20

修正前は、rubocopの`LineLength`を回避するために苦しい書き方をしたなと思っていたのですが、この修正でシンプルになったのでとても嬉しいです✨